clients/go/pkg/audit/interceptor.go

```diff
@@ -82,28 +82,21 @@ func WithInterceptorLogMode(m api.AuditLogRequest_LogMode) InterceptorOption {
 	}
 }
 
-// WithJWTValidator configures the interceptor to use the JWTValidator client to add justifications to logs.
-func WithJWTValidator(j JWTValidator) InterceptorOption {
+func WithJustification(j *justification.Processor) InterceptorOption {
 	return func(i *Interceptor) error {
-		i.jwtValidator = j
+		i.justProcessor = j
 		return nil
 	}
 }
 
-// JWTValidator is intended to provide a method for validating JWTs passed in using the
-// justification-token header.
-type JWTValidator interface {
-	ValidateJWT(jwtStr string) (*jwt.Token, error)
-}
-
 // Interceptor contains the fields required for an interceptor
 // to autofill and emit audit logs.
 type Interceptor struct {
 	*Client
-	sc           security.GRPCContext
-	rules        []*api.AuditRule
-	logMode      api.AuditLogRequest_LogMode
-	jwtValidator JWTValidator
+	sc            security.GRPCContext
+	rules         []*api.AuditRule
+	logMode       api.AuditLogRequest_LogMode
+	justProcessor *justification.Processor
```

Comment:
Nit: shortened names this are a bit weird. Can we just say "justificationProcessor"?

Reply:
+1